### ISSUE TITLE
Complex types binding

### ISSUE THREAD

**Issue by michalsznajder** [dotnet/AspNetCore.Docs#25091]
In section for `Complex types` I think this sentence 

> For each property of the complex type, model binding looks through the sources for the name pattern prefix.property_name. If nothing is found, it looks for just property_name without the prefix.

is not 100% correct.

If a query like `?Instructor.Id=100&Name=foo` is bound to method `void OnGet(Instructor instructor)` than resulting object of type `Instructor` will have field `Id` set to `100` but field `Name` will be `null`. 

Not sure if this for all targets but at parameters is is like that in ASP.NET 6.0 (cf. https://github.com/dotnet/aspnetcore/blob/ae1a6cbe225b99c0bf38b7e31bf60cb653b73a52/src/Mvc/Mvc.Core/src/ModelBinding/ParameterBinder.cs#L157)

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: e24f48f8-6c4d-ecda-6ce6-a7826d396e99
* Version Independent ID: 9c4af557-8d06-4c6c-ad2a-0dd234547563
* Content: [Model Binding in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/mvc/models/model-binding?view=aspnetcore-6.0#complex-types)
* Content Source: [aspnetcore/mvc/models/model-binding.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/mvc/models/model-binding.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-mvc**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

**Comment by michalsznajder** [comment]
From my investigation is was always like that (at least in ASP.NET Core) and I would treat this as documentation issue only.

When I was hit by this issue I first checked docs and they stated that `ID` would bind in this scenario. 4 hours later after a lot of debugging, code reading and testing I managed to find implementation. And it actually makes sense for me. It was only docs that put me into wrong way.
